tools/c7n_left/c7n_left/providers/terraform/graph.py

```diff
@@ -36,9 +39,65 @@ def get_resources_by_type(self, types=()):
                 yield type_name, resources
 
     def as_resource(self, name, data):
-        if isinstance(data["__tfmeta"], list):
-            for m in data["__tfmeta"]:
-                m["src_dir"] = self.src_dir
-        else:
-            data["__tfmeta"]["src_dir"] = self.src_dir
+        data["__tfmeta"]["src_dir"] = self.src_dir
         return TerraformResource(name, data)
+
+    def build(self):
+        self.resolver = Resolver()
+        self.resolver.build(self.resource_data)
+        return self.resolver
+
+    def get_refs(self, resource, target_type):
+        return self.resolver.resolve_refs(resource, (target_type,))
+
+
+class Resolver:
+    def __init__(self):
+        self._id_map = {}
+        self._ref_map = {}
+
+    @staticmethod
+    def is_id_ref(v):
```

Comment:
This function isn't necessary once we merge:

https://github.com/cloud-custodian/tfparse/pull/58

It adds `__ref__` to the value so its clear which ones are actual references.

Reply:
cool, that's a nice change up for tfparse as well, and its nice to get away from implicit string checks here.